eng/tools/generator/cmd/v2/automation/automationCmd.go

```diff
@@ -101,64 +102,137 @@ func (ctx *automationContext) generate(input *pipeline.GenerateInput) (*pipeline
 		return nil, fmt.Errorf("failed to get sdk repo: %+v", err)
 	}
 
-	if input.RelatedReadmeMdFile != "" {
-		input.RelatedReadmeMdFiles = append(input.RelatedReadmeMdFiles, input.RelatedReadmeMdFile)
-	}
+	// typespec
+	// Generated by tsp only when tspconfig.yaml exists and has typespec-go option
+	existTypeSpecProject := false
+	if len(input.RelatedTypeSpecProjectFolder) > 0 {
+		for _, tspProjectFolder := range input.RelatedTypeSpecProjectFolder {
+			tspconfigPath := filepath.Join(tspProjectFolder, "tspconfig.yaml")
+			tsc, err := typespec.ParseTypeSpecConfig(tspconfigPath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse tspconfig.yaml: %+v", err)
+			}
 
-	for _, readme := range input.RelatedReadmeMdFiles {
-		log.Printf("Start to process readme file: %s", readme)
+			if _, ok := tsc.TypeSpecProjectSchema.Options[string(typespec.TypeSpec_GO)]; ok {
+				log.Printf("Start to process typespec project: %s", tspProjectFolder)
+				existTypeSpecProject = true
+				generateCtx := common.GenerateContext{
+					SDKPath:        sdkRepo.Root(),
+					SDKRepo:        &sdkRepo,
+					SpecPath:       ctx.specRoot,
+					SpecCommitHash: ctx.commitHash,
+					TypeSpecConfig: tsc,
+				}
 
-		sepStrs := strings.Split(readme, "/")
-		for i, sepStr := range sepStrs {
-			if sepStr == "resource-manager" {
-				readme = strings.Join(sepStrs[i-1:], "/")
-				if i > 1 {
-					ctx.specRoot = input.SpecFolder + "/" + strings.Join(sepStrs[:i-1], "/")
+				module, err := tsc.GetModuleName()
+				if err != nil {
+					return nil, err
+				}
+
+				namespaceResult, err := generateCtx.GenerateForTypeSpec(&common.GenerateParam{
+					RPName:              module[0],
+					NamespaceName:       module[1],
+					SkipGenerateExample: true,
+					GoVersion:           ctx.goVersion,
```

Comment:
Why doesn't this call pass in the emitter options?

Reply:
automation-v2 cannot get emitter options